packages/tenant-process/src/services/tenantService.ts

```diff
@@ -1189,6 +1190,74 @@ export function tenantServiceBuilder(
       }
       return tenant.data;
     },
+
+    async m2mRevokeCertifiedAttribute({
+      organizationId,
+      tenantOrigin,
+      tenantExternalId,
+      attributeExternalId,
+      correlationId,
+      logger,
+    }: {
+      organizationId: TenantId;
+      tenantOrigin: string;
+      tenantExternalId: string;
+      attributeExternalId: string;
+      correlationId: string;
+      logger: Logger;
+    }): Promise<void> {
+      logger.info(`Revoking certified attribute to tenant`);
+      const requesterTenant = await retrieveTenant(
+        organizationId,
+        readModelService
+      );
+
+      const certifierId = requesterTenant.data.features.find(
+        (f) => f.type === "PersistentCertifier"
+      )?.certifierId;
+
+      if (!certifierId) {
+        throw tenantIsNotACertifier(requesterTenant.data.id);
+      }
+      const targetTenant = await retrieveTenantByExternalId({
+        tenantOrigin,
+        tenantExternalId,
+        readModelService,
+      });
+
+      const attributeToRevoke = await retrieveCertifiedAttribute({
+        attributeOrigin: certifierId,
+        attributeExternalId,
+        readModelService,
+      });
+
+      const attributePreviouslyAssigned = targetTenant.data.attributes.find(
+        (attr): attr is CertifiedTenantAttribute =>
+          attr.type === tenantAttributeType.CERTIFIED &&
+          attr.id === attributeToRevoke.id
+      );
+
+      if (!attributePreviouslyAssigned) {
+        throw attributeNotFoundInTenant(
+          attributeToRevoke.id,
+          targetTenant.data.id
+        );
+      }
+
+      const updatedTenant = await revokeCertifiedAttribute(
```

Comment:
Revoking a certified attribute could change the tenant's kind.
We should apply the tenant kind check and modify it here.

Reply:
I know, but I didn't find that in the Scala implementation of this endpoint

Reply:
Fixed, thanks